### PR TITLE
CLOUDP-245765: Add goreleaser to mongodb/openapi repo

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-245765

This PR adds goreleaser configuration to the openapi cli tool
I tested the goreleaser configuration in a fork and I was able to release the cli: https://github.com/andreaangiolillo/openapi-test/releases/tag/v0.0.1

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tools/cli/.goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Visit https://goreleaser.com for documentation on how to customize this
+# behavior.
+project_name: openapicli
```
Comment (gssbzn) [inline tools/cli/.goreleaser.yml:3]: we should come up with a better name for the cli

**Review by gssbzn — APPROVED** [review]
LGTM

File: tools/cli/.goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+archives:
+- id: linux_archives
+  name_template: mongodb-openapi-cli_{{ .Version }}_{{ .Os }}_{{- if eq .Arch "amd64" }}x86_64{{- else }}{{ .Arch }}{{ end }}
+  builds: [linux]
+  wrap_in_directory: true
+  format: tar.gz
+  rlcp: false
+- id: macos
+  name_template: mongodb-atlas-cli_{{ .Version }}_{{- if eq .Os "darwin" }}macos{{- else }}{{ .Os }}{{ end }}_{{- if eq .Arch "amd64" }}x86_64{{- else }}{{ .Arch }}{{ end }}
```
Comment (fmenezes) [inline tools/cli/.goreleaser.yml:43]: copy paste?

Comment (fmenezes) [reply tools/cli/.goreleaser.yml:43]:
```suggestion
  name_template: mongodb-openapi-cli_{{ .Version }}_{{- if eq .Os "darwin" }}macos{{- else }}{{ .Os }}{{ end }}_{{- if eq .Arch "amd64" }}x86_64{{- else }}{{ .Arch }}{{ end }}
```

File: tools/cli/.goreleaser.yml (hunk 1 above)
Comment (fmenezes) [reply tools/cli/.goreleaser.yml:3]: mongoapi?

File: tools/cli/.goreleaser.yml (hunk 2 above)
Comment (andreaangiolillo) [reply tools/cli/.goreleaser.yml]: updated. Thanks!

File: tools/cli/.goreleaser.yml (hunk 1 above)
Comment (andreaangiolillo) [reply tools/cli/.goreleaser.yml:3]: I opened a pool on the team slack channel, I will create a ticket to rename the cli after everyone has voted

**Pull request merged**